rust/abacus-base/src/inbox.rs

```diff
@@ -54,22 +54,24 @@ impl Inbox for CachingInbox {
     }
 }
 
-impl AbacusContract for CachingInbox {
+impl AbacusChain for CachingInbox {
     fn chain_name(&self) -> &str {
         self.inbox.chain_name()
     }
 
+    fn local_domain(&self) -> u32 {
+        self.inbox.local_domain()
+    }
+}
+
+impl AbacusContract for CachingInbox {
     fn address(&self) -> H256 {
         self.inbox.address()
     }
 }
 
 #[async_trait]
 impl AbacusCommon for CachingInbox {
-    fn local_domain(&self) -> u32 {
-        self.inbox.local_domain()
-    }
-
     async fn status(&self, txid: H256) -> Result<Option<TxOutcome>, ChainCommunicationError> {
```

Comment:
New to this code but this seems like a bizarre property to have on a contract, given that it's more or less independent of the contract itself, but instead is just using the RPC provider the contract object is using to send an RPC

Reply:
Not sure of a good way to change it without making a lot of other changes since we don't construct the pure provider instances outside the scraper at this time. Maybe it would make sense to also allow getting an `AbacusProvider` from any abacus contract or something and use that for status.

Ultimately this right now is here as part of the design legacy.

@tkporter thoughts?

Reply:
Yeah to be clear I'm not suggesting any specific changes rn just documenting my thoughts as I start to get up to speed on this codebase a bit

Reply:
You're totally right, this is super weird. git blame says we inherited this and it seems like the function isn't even called anywhere. I just commented out the function from the trait / all impls and everything compiles, so I think we can (and should) just remove this

Reply:
Done.